src/test/java/org/takes/misc/VerboseIteratorTest.java

```diff
@@ -0,0 +1,53 @@
+package org.takes.misc;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * Test case for {@link VerboseIterator}
+ * @author Zarko Celebic
+ */
+public final class VerboseIteratorTest{
+
+    @Rule
+    public ExpectedException expectedEx = ExpectedException.none();
+
+    @Test
+    public void testHasNext() throws Exception {
+        List<String> testList = Arrays.asList("1", "2", "3", "4", "5");
+        VerboseIterator<String> vi = new VerboseIterator<String>(testList.iterator(), "Error message");
+        MatcherAssert.assertThat(vi.hasNext(), Matchers.equalTo(true));
+        vi.next();
+        MatcherAssert.assertThat(vi.hasNext(), Matchers.equalTo(true));
+        vi.next();
+        MatcherAssert.assertThat(vi.hasNext(), Matchers.equalTo(true));
+        vi.next();
+        MatcherAssert.assertThat(vi.hasNext(), Matchers.equalTo(true));
+        vi.next();
+        MatcherAssert.assertThat(vi.hasNext(), Matchers.equalTo(true));
+        vi.next();
+        MatcherAssert.assertThat(vi.hasNext(), Matchers.equalTo(false));
+
+    }
+
+    @Test
+    public void testNext() throws Exception {
```

Comment:
@celezar see http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names for test name and test description.